sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -1022,6 +1022,209 @@ constexpr void check_atomic() {
+
+/// Store a scalar value into the Shared Local Memory.
+/// @tparam T type of the value
+/// @param offset SLM offset in bytes
+/// @param val value to store
+///
+template <typename T>
+__ESIMD_API void slm_scalar_store(uint32_t offset, T val) {
+  slm_scatter<T, 1>(simd<uint32_t, 1>(offset), simd<T, 1>(val), 1);
+}
+
+/// Gather data from the Shared Local Memory at specified \c offsets and
+/// return it as simd vector. See @ref usm_gather_rgba for information about
+/// the operation semantics and parameter restrictions/interdependencies.
+/// @tparam T The element type of the returned vector.
+/// @tparam N The number of elements to access.
+/// @tparam RGBAMask Pixel's channel mask.
+/// @param offsets Byte offsets within the SLM of each element.
+/// @param mask Operation mask. All-1 by default.
+/// @return Gathered data as an \c N - element vector.
+///
+template <typename T, int N, rgba_channel_mask RGBAMask>
+__ESIMD_API std::enable_if_t<(N == 8 || N == 16 || N == 32) && (sizeof(T) == 4),
+                             simd<T, N * get_num_channels_enabled(RGBAMask)>>
+slm_gather_rgba(simd<uint32_t, N> offsets, simd_mask<N> mask = 1) {
+  const auto SI = __ESIMD_NS::get_surface_index(detail::LocalAccessorMarker());
+  return __esimd_gather4_masked_scaled2<T, N, RGBAMask>(
+      SI, 0 /*global_offset*/, offsets.data(), mask.data());
+}
+
+/// Gather data from the Shared Local Memory at specified \c offsets and
+/// return it as simd vector. See @ref usm_scatter_rgba for information about
+/// the operation semantics and parameter restrictions/interdependencies.
+/// @tparam T The element type of the returned vector.
+/// @tparam N The number of elements to access.
+/// @tparam Mask Pixel's channel mask.
+/// @param offsets Byte offsets within the SLM of each element.
+/// @param vals values to be written.
+/// @param mask Operation mask. All-1 by default.
+///
+template <typename T, int N, rgba_channel_mask Mask>
+__ESIMD_API std::enable_if_t<(N == 8 || N == 16 || N == 32) && (sizeof(T) == 4)>
+slm_scatter_rgba(simd<uint32_t, N> offsets,
+                 simd<T, N * get_num_channels_enabled(Mask)> vals,
+                 simd_mask<N> mask = 1) {
+  detail::validate_rgba_write_channel_mask<Mask>();
+  const auto si = __ESIMD_NS::get_surface_index(detail::LocalAccessorMarker());
+  constexpr int16_t Scale = 0;
+  constexpr int global_offset = 0;
+  __esimd_scatter4_scaled<T, N, decltype(si), Mask, Scale>(
+      mask.data(), si, global_offset, offsets.data(), vals.data());
+}
+
+/// Loads a contiguous block of memory from the SLM at given offset and
+/// returns the loaded data as a vector.
+/// @tparam T Element type.
+/// @tparam N Number of elements to load, <code>N * sizeof(Tx)</code> must be
+///    1, 2, 4 or 8 owords long.
+/// @param offset The offset to load from in bytes. Must be oword-aligned.
+/// @return A vector of loaded elements.
+///
+template <typename T, int N>
+__ESIMD_API simd<T, N> slm_block_load(uint32_t offset) {
```

Comment:
This version of the function is obsolete and needs to be updated, please use the version from syclos/head.

Reply:
The function moved to here from L1841 had 'Flags' operand.